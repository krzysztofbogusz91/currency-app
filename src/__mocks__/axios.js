export default {
    get: jest.fn(() => Promise.resolve({
        data: [
            {
              rates: [
                {
                    "currency": "dolar ameryka≈Ñski",
                    "code": "USD",
                    "bid": 3.4034,
                    "ask": 3.4722
                    },
                    {
                    "currency": "dolar australijski",
                    "code": "AUD",
                    "bid": 2.5905,
                    "ask": 2.6429
                    }
              ]  //rates
            }
        ]
    }//end resolve ob
))

}