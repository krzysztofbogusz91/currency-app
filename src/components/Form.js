import React, { Component } from 'react';
import PropTypes from 'prop-types';


const propTypes = {
    
};


class Form extends Component {
    render() {
        return (
            <div className="mt-5">
                FORM
            </div>
        );
    }
}


Form.propTypes = propTypes;


export default Form;
