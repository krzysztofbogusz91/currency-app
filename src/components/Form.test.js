import React from 'react';
import {shallow, mount, render} from 'enzyme';
import { config } from '../enzyme_set_up';
import Form from './Form';
import { mockData } from 'axios';

describe('Form', () => {
    const form = shallow(<Form />)
    it('renders properly', ()=>{
        expect(form).toMatchSnapshots();
    })
});