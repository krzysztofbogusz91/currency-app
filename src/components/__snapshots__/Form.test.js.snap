// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "FORM",
      "className": "mt-5",
    },
    "ref": null,
    "rendered": "FORM",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "FORM",
        "className": "mt-5",
      },
      "ref": null,
      "rendered": "FORM",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
